### PR TITLE
Fallback to pytorch engine when the model is quantized by smooth quant

### PR DESCRIPTION
**Pull request by lvhan028**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by lvhan028** [comment]
cc @zhulinJulia24

**Comment by zhulinJulia24** [comment]
related to https://github.com/InternLM/lmdeploy/issues/2595

I need update testcase in this pr and test it.
image

File: lmdeploy/archs.py [hunk 1]
```diff
@@ -193,3 +193,21 @@ def get_model_arch(model_path: str):
...
+    """get quantization config from a model's config."""
+    if isinstance(config, dict):
+        for k, v in config.items():
+            if k == 'quantization_config':
+                return v
+            if isinstance(v, (dict, list)):
+                result = get_quantization_config(v)
+                if result is not None:
+                    return result
+    elif isinstance(config, list):
```
Comment (AllentDan) [inline lmdeploy/archs.py:208]: Is it possible that `quantization_config` is inside a list? Seems `find_quantization_config(nested, target_key)` is a more general implementation.

Comment (lvhan028) [reply lmdeploy/archs.py:208]: `get_quantization_config` is implanted from `find_quantization_config` except that `target_key` is hardcoded 'quantization_config`

Comment (AllentDan) [reply lmdeploy/archs.py:208]: Yes, I mean the original function is a general function that we can also get other configurations through `target_key` such as `llm_config` or `language_config`.

Comment (lvhan028) [reply lmdeploy/archs.py:208]: I see. In that case, I'd better change the function name, too

**Review by AllentDan — APPROVED** [review]
Tested OK with internlm2-chat-1_8b-w8a8

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by RunningLeon — COMMENTED** [review]
tested ok on Meta-Llama-3-8B-Instruct with bf16, int8-w8a8, awq.

**Pull request merged**